### PR TITLE
NT-0031 - Changed JWT token Authenticate with passport/passport-jwt

### PR DESCRIPTION
**Pull request by kamilmoskal**

### CODE REVIEW

File: controllers/users/users.js [hunk 1]
```diff
@@ -40,58 +42,80 @@ exports.activateUser = (req, res) => {
...
-      newUser.save()
+      newUser
+        .save()
         .then(user => {
-          const payload = { 
-            user: { 
-              id: user.id 
-            } 
+          const payload = {
+            user: {
```
Comment (damian0o) [inline controllers/users/users.js]: User id can be stored in subject field

{
 sub: user.id
}

Maybe passport js provide us with some support for creation of JWT as well

File: controllers/users/users.js [hunk 2]
```diff
@@ -12,24 +12,26 @@ const sendVerificationEmail = user => {
...
   const mailOptions = {
     from: "natripareact <natripareact@gmail.com>",
     to: user.email,
     subject: "Account Activation",
     html: `
       To activate your account, click this link
-      <a href="http://localhost:5000/api/users/activate/${user.activationKey}">Activate Account</a>
+      <a href="http://localhost:5000/api/users/activate/${
+        user.activationKey
+      }">Activate Account</a>
```
Comment (mras2303) [inline controllers/users/users.js]: Completely unreadable for me if there will be more variables in template literals

File: middleware/auth.js [hunk 3]
```diff
@@ -1,32 +1,29 @@
...
-          req.user = decoded.user;
-          next();
-        } else {
-          res.status(401).json({ msg: messages.EMAIL_NOT_VERIFIED });
-        }
-      });
+      return res.status(500).json({ msg: messages.SERVER_ERROR });
+    }
+    if (info !== undefined) {
+      if (info.message === "No auth token") {
```
Comment (mras2303) [inline middleware/auth.js]: Move it to messages file

File: middleware/auth.js [hunk 4]
```diff
@@ -1,32 +1,29 @@
...
-        if (user.emailVerified) {
-          req.user = decoded.user;
-          next();
-        } else {
-          res.status(401).json({ msg: messages.EMAIL_NOT_VERIFIED });
-        }
-      });
+      return res.status(500).json({ msg: messages.SERVER_ERROR });
+    }
+    if (info !== undefined) {
```
Comment (mras2303) [inline middleware/auth.js]: is that condition need?

```if(info)```

should make the same

File: middleware/auth.js [hunk 5]
```diff
@@ -1,32 +1,29 @@
...
+      if (info.message === "No auth token") {
+        return res
+          .status(401)
+          .json({ msg: messages.NO_TOKEN_AUTHORIZATION_DENIED });
+      } else if (
+        info.message === "invalid signature" ||
```
Comment (mras2303) [inline middleware/auth.js]: Move it to messages file

File: middleware/auth.js [hunk 6]
```diff
@@ -1,32 +1,29 @@
...
+        info.message === "jwt malformed"
```
Comment (mras2303) [inline middleware/auth.js]: Move it to messages file

File: middleware/auth.js [hunk 7]
```diff
@@ -1,32 +1,29 @@
...
+      ) {
+        return res.status(401).json({ msg: messages.TOKEN_IS_INVALID });
+      } else if (info.message === "jwt expired") {
```
Comment (mras2303) [inline middleware/auth.js]: Move it to messages file

File: middleware/auth.js [hunk 8]
```diff
@@ -1,32 +1,20 @@
...
-          req.user = decoded.user;
-          next();
-        } else {
-          res.status(401).json({ msg: messages.EMAIL_NOT_VERIFIED });
-        }
+      return res.status(500).json({ msg: messages.SERVER_ERROR });
+    }
+    if (info) {
+      return res.status(401).json({
+        msg: info.message[0].toUpperCase() + info.message.slice(1)
```
Comment (mras2303) [inline middleware/auth.js]: Read about array destructuring

**Pull request merged**